akita/src/internal/crud.ts

```diff
@@ -92,15 +104,25 @@ export class CRUD {
         newEntity = new oldEntity.constructor(merged);
       }
 
-      updatedEntities[id] = newEntity;
+      updatedEntities[idToUpdate] = newEntity;
+    }
+
+    let updatedIds = state.ids;
+    let stateEntities = state.entities;
+    if (isUpdatingIdKey) {
+      const id = ids[0];
```

Comment:
you can destruct directly `const [id] = ids;`

Reply:
will do! Not used to this magic yet :)